### PR TITLE
Past due positions

### PR DESCRIPTION
**Pull request by dnut**
If a non-zero position is past due, it must be repaid immediately, which means the account should be marked as unhealthy so a liquidator can force repayment.

### CODE REVIEW

**Review by jetjake — COMMENTED** [review]
Probably a stupid question, but we don't need a new metadata account for the token kind to have the value of a "PastDueClaim" such as somewhere in here

https://github.com/jet-lab/jet-v2/blob/master/programs/control/src/instructions/register_token.rs#L206

File: programs/margin/src/instructions/register_position.rs [hunk 1]
```diff
@@ -66,21 +66,27 @@ pub fn register_position_handler(ctx: Context<RegisterPosition>) -> Result<()> {
...
-        kind,
+        metadata.token_kind.into(),
         metadata.collateral_weight,
         metadata.collateral_max_staleness,
     )?;
 
     Ok(())
 }
+
+impl From<TokenKind> for PositionKind {
```
[the remark this answers is not in the document]
Comment (dnut) [reply programs/margin/src/instructions/register_position.rs]: My guess is that it enables us to have a mapping in the future that isn't 1-to-1. It's like a form of encapsulation between the different programs so they aren't bound to internal representations that depend on another program's data structures.

I'm puzzled as to why we use different names e.g. Collateral vs Deposit.

@suscd any thoughts?

**Comment by dnut** [comment]
> Probably a stupid question, but we don't need a new metadata account for the token kind to have the value of a "PastDueClaim" such as somewhere in here
> 
> https://github.com/jet-lab/jet-v2/blob/master/programs/control/src/instructions/register_token.rs#L206

Yes but not because there's a new token kind. We're going to need significant changes in control to support any adapters beside pools that can issue collateral or claims. This instruction is too opinionated that tokens = pools and it needs to be broken into several smaller instructions that can be composed as needed by clients.

File: programs/margin/src/instructions/register_position.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (suscd) [reply programs/margin/src/instructions/register_position.rs]: Not sure why the names are different, was wondering the same thing myself. We could fix that

**Review by suscd — DISMISSED** [review]

**Pull request merged**